### ISSUE TITLE
Remove Doublicate Items From Array

### ISSUE THREAD

**Issue by IamDevelish** [ocodia/Check-My-Links#30]
Hi, 

I absolutely love this extension... BUT

The extension seems to check doublicate URLs / Links which are on a given page - it would be handy to have a option / feature that turns this off/on

Thanks a LOT

mariano

**Comment by ocodia** [comment]
Good idea, Mariano, I'll keep that in mind when I'm doing the next round of improvements. In the mean time you can enable caching on the options page of the extension, and that will prevent it from checking links which are known to be working ok (which should speed up page checks significantly). 

--
Paul Livingstone

**Comment by IamDevelish** [comment]
for a quick fix for me i just added this to the file "check.js" at round line 190

          else{
          queued +=1;
blacklist.push(url); // added this line
          link.classList.add("CMY_Link");
          checkURL(url, link);
          }

I add an item to the blacklist array - this way a url is counted and checked only once

**Comment by kyleladd** [comment]
Good idea,

To continue down that path, let's say there is a 404 link (http://notreal.ex) in multiple places within the DOM. By only checking the link once the classes `link.classList.add("CMY_Valid");` or `link.classList.add("CMY_Invalid");` would only be added to one of the links.

As a result, there would need to be a modification to the `updateDisplay` function. `updateDisplay` would need to be passed `pageLinks` array.  `updateDisplay` would then need to cycle through the `pageLinks` array and add the corresponding class to all elements with the checked `url`.

-Kyle

**Comment by IamDevelish** [comment]
Could you please include a dublicate url check in the current / future release? When I edited the files Chrome warned me that the plugin is not valid / is broken.

**Comment by kyleladd** [comment]
This can be done. Filter [pageLinks](https://github.com/ocodia/Check-My-Links/blob/master/check.js#L21) to only unique based on the link.href attribute.  Just make sure that when [updating the UI](https://github.com/ocodia/Check-My-Links/blob/master/functions.js#L152), update all instances of the link href in the DOM with the linkStatus and warnings.

**Comment by IamDevelish** [comment]
It would be great if it could be an option in the config screen - and in the main branch :-)

**Comment by gogoanimeHDmobi** [comment]
that is good work keep working i am immpressed thakns
